include/envoy/http/hash_policy.h

```diff
@@ -2,12 +2,19 @@
 
 #include "envoy/http/header_map.h"
 #include "envoy/network/address.h"
+#include "envoy/stream_info/filter_state.h"
 
 #include "absl/types/optional.h"
 
 namespace Envoy {
 namespace Http {
 
+class Hashable {
+public:
+  virtual absl::optional<uint64_t> hash() const { return absl::nullopt; };
```

Comment:
Why isn't this `PURE`? It seems that any hashable object should be able to always produce a hash?

Reply:
Removed.